src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/RequireEmptyLineBeforeAtClauseBlockCheck.java

```diff
@@ -0,0 +1,201 @@
+     *  -- WS
+     *  -- JAVADOC_TAG
+     * </pre>
+     */
+    private static final int NUM_OF_SIBLINGS_BEFORE_ONLY_JAVADOC_TAG = 3;
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC_TAG,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode ast) {
+        // No need to filter token because overridden getDefaultJavadocTokens ensures that we only
+        // receive JAVADOC_TAG DetailNode.
+        checkAtClause(ast);
+    }
+
+    /**
+     * Logs when there is no empty line before the at clause.
+     *
+     * @param atClause the at clause to check for an empty space before it.
+     */
+    private void checkAtClause(DetailNode atClause) {
+        if (!isAnotherAtClauseBefore(atClause)
+                && !isOnlyAtClauseNothingElse(atClause)
+                && insufficientConsecutiveNewlines(atClause)) {
+            log(atClause.getLineNumber(),
+                    MSG_JAVADOC_TAG_LINE_BEFORE,
+                    atClause.getChildren()[0].getText());
+        }
+    }
+
+    private static boolean isAnotherAtClauseBefore(DetailNode atClause) {
+        boolean found = false;
+        DetailNode currentClause = JavadocUtil.getPreviousSibling(atClause);
+        while (currentClause != null) {
+            if (currentClause.getType() == JavadocTokenTypes.JAVADOC_TAG) {
+                found = true;
+                break;
+            }
+            currentClause = JavadocUtil.getPreviousSibling(currentClause);
+        }
+        return found;
+    }
+
+    private static boolean isOnlyAtClauseNothingElse(DetailNode atClause) {
+        int count = 0;
+        DetailNode currentClause = JavadocUtil.getPreviousSibling(atClause);
+        while (currentClause != null) {
+            count++;
+            currentClause = JavadocUtil.getPreviousSibling(currentClause);
+        }
+        return count == NUM_OF_SIBLINGS_BEFORE_ONLY_JAVADOC_TAG;
+    }
+
+    /**
+     * Iterates through the previous siblings of the atClause looking for empty lines until
+     * there are no more siblings or it hits something other than asterisk, whitespace or newline.
+     * If it finds at least one empty line, return true. Return false otherwise.
+     *
+     * @param atClause the atClause to check if there are sufficient empty lines before it.
+     * @return true if there are enough empty lines before the atClause.
+     */
+    private static boolean insufficientConsecutiveNewlines(DetailNode atClause) {
+        int count = 0;
+        DetailNode currentClause = JavadocUtil.getPreviousSibling(atClause);
+        while (currentClause != null
+                && count <= 1
+                && (currentClause.getType() == JavadocTokenTypes.NEWLINE
+                || currentClause.getType() == JavadocTokenTypes.WS
```

Comment:
Please indent this and following lines further to right, to visually show nested expression.

Reply:
I added an additional indentation to both the || lines.